custom/icds_reports/management/commands/custom_ls_data_pull.py

```diff
@@ -0,0 +1,153 @@
+import csv
+
+from corehq.apps.es.forms import FormES
+from datetime import datetime
+from corehq.apps.users.models import CommCareUser
+from django.core.management.base import BaseCommand
+from django.db import connections
+
+from corehq.sql_db.connections import get_icds_ucr_citus_db_alias
+
+
+def _run_custom_sql_script(command, day=None):
+    db_alias = get_icds_ucr_citus_db_alias()
+    if not db_alias:
+        return
+
+    cursor = connections[db_alias].cursor()
+    cursor.execute(command, [day])
+
+    return cursor.fetchall()
+
+
+def state_details():
+    query = """
+        SELECT
+            awc.state_name as state_name,
+            awc.supervisor_name as supervisor_name,
+            l.supervisor_id as supervisor_id,
+            awc.supervisor_site_code as supervisor_site_code,
+            CASE WHEN l.num_supervisor_launched>0 THEN 'Launched'
+            ELSE 'UnLaunched' END as supervisor_launched_status
+            FROM "agg_ls" l
+            LEFT JOIN "awc_location_local" awc
+                ON (
+                awc.supervisor_id = l.supervisor_id
+                AND l.aggregation_level=awc.aggregation_level )
+            WHERE l.month='2020-04-01' AND l.aggregation_level=4 AND awc.supervisor_is_test<>1
```

Comment:
instead of supervisor is test<>1 good to use state_is_test<>1 as if any location is test it will under test state

Reply:
updated!